web/modules/weather_data/src/Service/WeatherAlertTrait.php

```diff
@@ -57,27 +57,65 @@ public static function tryParsingDescriptionText($str)
     /**
      * Get active alerts for a WFO grid cell.
      */
-    public function getAlertsForGrid($wfo, $x, $y)
+    public function getAlertsForGrid($wfo, $x, $y, $self = false)
     {
-        $geometry = $this->getGeometryFromGrid($wfo, $x, $y);
-        $point = $geometry[0];
+        if (!$self) {
+            $self = $this;
+        }
 
-        return $this->getAlertsForLatLon($point->lat, $point->lon);
-    }
+        $CACHE_KEY = "alerts $wfo/$x/$y";
+        $cache = $this->cache->get($CACHE_KEY);
+        if ($cache) {
+            return $cache->data;
+        }
 
-    /**
-     * Get active alerts for a latitude and longitude.
-     */
-    public function getAlertsForLatLon($lat, $lon)
-    {
-        $lat = round($lat, 4);
-        $lon = round($lon, 4);
+        $geometry = $self->getGeometryFromGrid($wfo, $x, $y);
+        $place = $self->getPlaceFromGrid($wfo, $x, $y);
+        $timezone = $place->timezone;
+
+        $alerts = $self->getFromWeatherAPI(
+            "/alerts/active?status=actual&area=$place->state",
+        )->features;
+
+        $geometry = array_map(function ($point) {
+            return $point->lon . " " . $point->lat;
+        }, $geometry);
+        $geometry = implode(",", $geometry);
+
+        $alerts = array_filter($alerts, function ($alert) use (
+            $place,
+            $geometry,
+        ) {
+            if (AlertPriority::isMarineAlert($alert->properties->event)) {
+                return false;
+            }
```

Comment:
Previously we removed marine alerts after transforming them. However, since we now do a little bit of math on the alerts, it makes sense to me to filter them out as early as possible.